spec/index.bs

```diff
@@ -6659,19 +6697,23 @@ dictionary GPUQuerySetDescriptor : GPUObjectDescriptorBase {
 };
 </script>
 
-<dl dfn-type=dict-member dfn-for=GPUQuerySetDescriptor>
-    : <dfn>pipelineStatistics</dfn>
-    ::
-        The set of {{GPUPipelineStatisticName}} values in this sequence defines which pipeline statistics will be returned in the new query set.
+- {{GPUQuerySetDescriptor/type}} specifies the type of queries managed by {{GPUQuerySet}}.
+- {{GPUQuerySetDescriptor/count}} specifies the number of queries managed by {{GPUQuerySet}}.
+- {{GPUQuerySetDescriptor/pipelineStatistics}} is a set of {{GPUPipelineStatisticName}} values to
+    determine which pipeline statistics will be returned in {{GPUQuerySet}}.
 
-        <div class=validusage dfn-for=GPUQuerySetDescriptor.pipelineStatistics>
-            <dfn abstract-op>Valid Usage</dfn>
+<div class=validusage dfn-for=GPUQuerySetDescriptor>
+    <dfn abstract-op>GPUQuerySetDescriptor Valid Usage</dfn>
 
-            1. |pipelineStatistics| is ignored if type is not {{GPUQueryType/pipeline-statistics}}.
-            2. If {{GPUFeatureName/pipeline-statistics-query}} is not available, |type| must not be {{GPUQueryType/pipeline-statistics}}.
-            3. If |type| is {{GPUQueryType/pipeline-statistics}},  |pipelineStatistics| must be a sequence of {{GPUPipelineStatisticName}} values which cannot be duplicated.
-        </div>
-</dl>
+    Given a {{GPUQuerySetDescriptor}} |this|, the following validation rules apply:
+
+    - |this|.{{GPUQuerySetDescriptor/type}} must not be {{GPUQueryType/pipeline-statistics}} if {{GPUFeatureName/pipeline-statistics-query}}
+        is not available.
+    - |this|.{{GPUQuerySetDescriptor/type}} must not be {{GPUQueryType/timestamp}} if {{GPUFeatureName/timestamp-query}} is not available.
+    - |this|.{{GPUQuerySetDescriptor/count}} &le; 8192.
+    - |this|.{{GPUQuerySetDescriptor/pipelineStatistics}} must be undefined if |this|.{{GPUQuerySetDescriptor/type}} is not
+        {{GPUQueryType/pipeline-statistics}}, otherwise it must be a sequence of {{GPUPipelineStatisticName}} values which cannot be duplicated.
+</div>
```

Comment:
Hm, I guess we should move these valid usage rules inside the definition of `createQuerySet` to match other parts of the spec, rather than keeping it with this descriptor.

(But maybe we should reconsider that editorially? Maybe it's better for readability if the rules for a descriptor are right next to the definition of the descriptor. @kvark @toji)

Reply:
Agree. It's better to move descriptor valid usage inside the definition of `createQuerySet` where the descriptor is used to do the validations.

Reply:
Discussed this at the editors meeting and realized that the descriptor is right next to createQuerySet anyway, so there is no readability concern with inlining the rules in createQuerySet. Resolved :)